tests/remote/remote_controller.pm

```diff
@@ -17,39 +17,51 @@ use testapi;
 use utils;
 use mm_network;
 use lockapi;
+use mmapi;
 
 sub run {
-    my ($self) = @_;
     my $target_ip;
+    my $children = get_children();
+    my $child_id = (keys %$children)[0];
+    my $password = 'nots3cr3t';
 
-    select_console 'root-console';
-    ensure_serialdev_permissions;
-
-    # Setup static NETWORK
-    configure_default_gateway;
-    configure_static_ip('10.0.2.12/15');
+    mutex_lock("dhcp_client_configuration", $child_id);
+    mutex_unlock("dhcp_client_configuration", $child_id);
+    mutex_lock("ssh_server_is_available", $child_id);
 
     # Wait until target becomes ready
-    mutex_lock "installation_ready";
+    # parse dhcpd.leases - now it should contain entries for all nodes
+    my %dhcp_leases;
+    my $dhcp_leases_file = script_output("cat /var/lib/dhcp/db/dhcpd.leases\n");
+    my $lease_ip;
 
-    if (check_var("REMOTE_CONTROLLER", "vnc")) {
-        # Get target ip using slptool
-        systemctl 'stop ' . $self->firewall;
-        $target_ip = script_output "slptool findsrvs service:YaST.installation.suse:vnc | cut -d: -f4 | tr -d /";
-        systemctl 'start ' . $self->firewall;
+    for my $line (split /\n/, $dhcp_leases_file) {
+        if ($line =~ /^lease\s+\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}/) {
+            (undef, $lease_ip) = split(/\s/, $line);
+        }
+        elsif ($line =~ /client-hostname\s+"(.*)"/) {
+            my $hostname = lc($1);
+            $dhcp_leases{$hostname} //= [];
+            push(@{$dhcp_leases{$hostname}}, $lease_ip);
+        }
+    }
+
+    mutex_unlock("ssh_server_is_available", $child_id);
+    select_console 'root-console';
```

Comment:
As you said, you select user-console later, so this might not be needed

Reply:
Agreed, root console is not mandatory.
http://dhcp228.suse.cz/tests/266#